app/Controllers/categoryController.php

```diff
@@ -97,14 +97,16 @@ public function updateAction(): void {
 				Minz_Request::bad(_t('feedback.sub.category.no_name'), $url_redirect);
 			}
 
-			if ($catDAO->searchById($id) == null) {
+			$cat = $catDAO->searchById($id);
+			if ($cat === null) {
 				Minz_Request::bad(_t('feedback.sub.category.not_exist'), $url_redirect);
 			}
 
-			$cat = new FreshRSS_Category($name);
-			$values = array(
+			$values = [
 				'name' => $cat->name(),
-			);
+				'kind' => $cat->kind(),
+				'attributes' => $cat->attributes(),
```

Comment:
We had a bug there, which was erasing the values of other columns when updating the name of a category.
I am not sure this function is used at all though (the one in `FreshRSS_subscription_Controller::categoryAction()` is used) - to be checked